app/assets/javascripts/discourse/controllers/grant-badge.js.es6

```diff
@@ -0,0 +1,87 @@
+import computed from "ember-addons/ember-computed-decorators";
+import { extractError } from 'discourse/lib/ajax-error';
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import Badge from 'discourse/models/badge';
+import UserBadge from 'discourse/models/user-badge';
+
+export default Ember.Controller.extend(ModalFunctionality, {
+  topicController: Ember.inject.controller("topic"),
+  loading: true,
+  saving: false,
+  selectedBadgeId: null,
+  allBadges: [],
+  userBadges: [],
+
+  @computed('topicController.selectedPosts')
+  post() {
+    return this.get('topicController.selectedPosts')[0];
+  },
+
+  @computed('post')
+  badgeReason(post) {
+    const url = post.get('url');
```

Comment:
This can be simplified since there is no need to add protocol and host to the url. 

In the server, the url is only used to extract the post and it will work as it is (it is only interested in the path).

Reply:
Another change in my PR is that I was closing the dialog automatically right after granting the badge (unless there was an error). Instead of the flash message, I was showing a spinner while in the  "saving" state, and closing it right after, like in the "set topic timer" form. Do we want to keep it open after clicking "Grant"? If we don't, you can get rid of some more code there too.

In any case, good job here @xrav3nz 👏, I was just pointing out some differences in my PR in case we can use some of it before it's lost, but yours looks 👌.

Reply:
Thanks ❤️ 

> In the server, the url is only used to extract the post and it will work as it is

Interestingly enough, even if only given a path, `URI.parse` does not raise an exception and produces a `URI::Generic` object. But since the API expects a full URL, I think it's better to send in a full URL in case of backend implementation changes?

>  closing the dialog automatically right after granting the badge

I was a bit torn on how the modal should behave too! 

IMHO closing the modal after setting a timer / changing owner is fine because the post itself will reflect the changes, on other hand, closing the modal after granting badges _may_ lead to confusions.

But I do agree that the modal does not necessarily need to stay open after a successful grant, other than for providing visual feedback.